lib/graphql/account_settings/cloud_object.dart

```diff
@@ -0,0 +1,120 @@
+import 'dart:async';
+
+import 'package:autonomy_flutter/common/injector.dart';
+import 'package:autonomy_flutter/database/cloud_database.dart';
+import 'package:autonomy_flutter/graphql/account_settings/account_settings_client.dart';
+import 'package:autonomy_flutter/graphql/account_settings/account_settings_db.dart';
+import 'package:autonomy_flutter/graphql/account_settings/cloud_object/address_cloud_object.dart';
+import 'package:autonomy_flutter/graphql/account_settings/cloud_object/connection_cloud_object.dart';
+import 'package:autonomy_flutter/util/migration/migration_util.dart';
+import 'package:package_info_plus/package_info_plus.dart';
+
+class CloudObjects {
+  late final String _deviceUniqueId;
+
+  late final WalletAddressCloudObject _walletAddressObject;
+  late final AccountSettingsDB _addressAccountSettingsDB;
+
+  late final ConnectionCloudObject _connectionObject;
+  late final AccountSettingsDB _connectionAccountSettingsDB;
+
+  late final AccountSettingsDB _settingsDataDB;
+
+  CloudObjects() {
+    unawaited(_init());
+  }
+
+  Future<String> _getBackupId() async {
+    PackageInfo packageInfo = await PackageInfo.fromPlatform();
+    String? deviceId = await MigrationUtil.getBackupDeviceID();
+
+    return '${deviceId}_${packageInfo.packageName}';
+  }
+
+  Future<void> _init() async {
+    _deviceUniqueId = await _getBackupId();
+
+    /// Wallet Address
+    _addressAccountSettingsDB = AccountSettingsDBImpl(
+        injector(), [_db, _commonKeyPrefix, _walletAddressKeyPrefix].join('.'));
+    _walletAddressObject = WalletAddressCloudObject(_addressAccountSettingsDB);
+
+    /// Connection
+    _connectionAccountSettingsDB = AccountSettingsDBImpl(
+        injector(), [_db, _deviceUniqueId, _connectionKeyPrefix].join('.'));
+    _connectionObject = ConnectionCloudObject(_connectionAccountSettingsDB);
+
+    /// Settings
+    _settingsDataDB = AccountSettingsDBImpl(injector(),
```

Comment:
Since we know exactly the keys on settings, the download could be just to download the keys, not to search.